File: 2022/C++/Day-2/Part-1.cpp
#include <string>
#include <fstream>
#include <iostream>
#include <unordered_map>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::ifstream inputFile("input.txt");
    //-------------------------------------
    int score(0);
    {
        std::unordered_map<std::string, int> outcomes = {
            {"A X", 3}, {"B X", 0}, {"C X", 6},
            {"A Y", 6}, {"B Y", 3}, {"C Y", 0},
            {"A Z", 0}, {"B Z", 6}, {"C Z", 3}
        };
        
        std::string round;
        while (std::getline(inputFile, round)) {
            if (round.empty()) break;
            switch (round.at(2)) {
                case 'X': score += 1; break;
                case 'Y': score += 2; break;
                case 'Z': score += 3; break;
            }
            score += outcomes[round];
        }
    }
    std::cout << score;
    //-------------------------------------
    return 0;
}

File: 2022/Python/Day-2/Part-1.py
from helper_functions import get_relative_path


outcomes = {
    "A X": 3, "B X": 0, "C X": 6,
    "A Y": 6, "B Y": 3, "C Y": 0,
    "A Z": 0, "B Z": 6, "C Z": 3,
}

scores = {'X': 1, 'Y': 2, 'Z': 3}


def main():
    score = 0
    
    with open(get_relative_path("input.txt")) as file:
        while Round := file.readline().strip():
            score += scores[Round[2]] + outcomes[Round]
            
    print(score)


if __name__ == "__main__":
    main()
